phoenix-core/src/it/java/org/apache/phoenix/end2end/DecodeBinaryFunctionIT.java

```diff
@@ -0,0 +1,202 @@
+                "SELECT DECODE_BINARY('\\\\x48\\\\x65\\\\x6C\\\\x6C\\\\x6F\\\\x50" +
+                        "\\\\x68\\\\x6F\\\\x65\\\\x6E\\\\x69\\\\x78', " + "'HBASE') FROM " + testTable);
+        assertTrue(rs.next());
+        byte[] actualBytes = rs.getBytes(1);
+        assertTrue(Arrays.equals("HelloPhoenix".getBytes(), actualBytes));
+    }
+
+    @Test
+    public void testDecodeHBaseFromHexLiteral() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+
+        String testTable = generateUniqueName();
+        String ddl = "CREATE TABLE " + testTable + " (id INTEGER PRIMARY KEY, data VARCHAR)";
+
+        conn.createStatement().execute(ddl);
+
+        PreparedStatement ps = conn.prepareStatement("UPSERT INTO " + testTable + " (id, data) VALUES (1, 'data')");
+        ps.execute();
+        conn.commit();
+
+        ResultSet rs = conn.createStatement().executeQuery(
+                "SELECT DECODE_BINARY('\\x48\\x65\\x6C\\x6C\\x6F\\x50\\x68\\x6F\\x65\\x6E\\x69\\x78', " +
+                        "'HBASE') FROM " + testTable);
+        assertTrue(rs.next());
+        byte[] actualBytes = rs.getBytes(1);
+        assertTrue(Arrays.equals("HelloPhoenix".getBytes(), actualBytes));
+    }
+
+    @Test
+    public void testInvalidDecodingFormat() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+        String testTable = generateUniqueName();
+        String ddl = "CREATE TABLE " + testTable + " (id INTEGER PRIMARY KEY, data VARCHAR)";
+
+        conn.createStatement().execute(ddl);
+
+        try {
+            conn.createStatement().executeQuery("SELECT DECODE_BINARY(data, 'INVALIDFORMAT') FROM " + testTable);
+            fail("Expected an exception for invalid encoding format");
+        } catch (SQLException e) {
+            assertEquals(SQLExceptionCode.TYPE_MISMATCH.getErrorCode(), e.getErrorCode());
+        }
+    }
+
+    @Test
+    public void testNullAndEmptyStringDecoding() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+        String testTable = generateUniqueName();
+        String ddl = "CREATE TABLE " + testTable + " (id INTEGER PRIMARY KEY, data VARCHAR)";
+
+        conn.createStatement().execute(ddl);
+
+        ResultSet rs = conn.createStatement().executeQuery("SELECT DECODE_BINARY(data, 'HBASE') FROM " + testTable);
+        assertFalse(rs.next());
+
+        ResultSet rs2 = conn.createStatement().executeQuery("SELECT DECODE_BINARY(NULL, 'HEX') FROM " + testTable);
+        assertFalse(rs2.next());
+    }
+
+    @Test
+    public void testLongBase64Decoding() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+
+        String testTable = generateUniqueName();
+        String ddl = "CREATE TABLE " + testTable + " (id INTEGER PRIMARY KEY, data VARCHAR)";
+        conn.createStatement().execute(ddl);
+
+        String base64Chunk = "SGVsbG9QaG9lbml4";
```

Comment:
This works because we do not disable padding.

Reply:
You're right. I will also add a separate test case to verify Base64 decoding with padding.